shell/platform/embedder/tests/embedder_unittests.cc

```diff
@@ -510,5 +513,560 @@ TEST_F(EmbedderTest, VMAndIsolateSnapshotSizesAreRedundantInAOTMode) {
+                      const std::string& name,
+                      sk_sp<SkImage> image) {
+  if (!image) {
+    return false;
+  }
+
+  auto data = image->encodeToData(SkEncodedImageFormat::kPNG, 100);
+
+  if (!data) {
+    return false;
+  }
+
+  fml::NonOwnedMapping mapping(static_cast<const uint8_t*>(data->data()),
+                               data->size());
+  return WriteAtomically(directory, name.c_str(), mapping);
+}
+
+//------------------------------------------------------------------------------
+/// Test the layer structure and pixels rendered when using a custom compositor.
+///
+TEST_F(EmbedderTest, CompositorMustBeAbleToRenderKnownScene) {
+  auto& context = GetEmbedderContext();
+
+  context.SetupCompositor();
+
+  context.GetCompositor().SetRenderTargetType(
+      EmbedderTestCompositor::RenderTargetType::kOpenGLTexture);
+
+  fml::CountDownLatch latch(6);
+
+  sk_sp<SkImage> scene_image;
+  context.SetNextSceneCallback([&](sk_sp<SkImage> scene) {
+    scene_image = std::move(scene);
+    latch.CountDown();
+  });
+
+  context.GetCompositor().SetNextPresentCallback(
+      [&](const FlutterLayer** layers, size_t layers_count) {
+        ASSERT_EQ(layers_count, 5u);
+
+        // Layer Root
+        {
+          FlutterBackingStore backing_store = *layers[0]->backing_store;
+          backing_store.type = kFlutterBackingStoreTypeOpenGL;
+          backing_store.did_update = true;
+          backing_store.open_gl.type = kFlutterOpenGLTargetTypeTexture;
+
+          FlutterLayer layer = {};
+          layer.struct_size = sizeof(layer);
+          layer.type = kFlutterLayerContentTypeBackingStore;
+          layer.backing_store = &backing_store;
+          layer.size = FlutterSizeMake(800.0, 600.0);
+          layer.offset = FlutterPointMake(0.0, 0.0);
+
+          ASSERT_EQ(*layers[0], layer);
+        }
+
+        // Layer 1
+        {
+          FlutterPlatformView platform_view = {};
+          platform_view.struct_size = sizeof(platform_view);
+          platform_view.identifier = 1;
+
+          FlutterLayer layer = {};
+          layer.struct_size = sizeof(layer);
+          layer.type = kFlutterLayerContentTypePlatformView;
+          layer.platform_view = &platform_view;
+          layer.size = FlutterSizeMake(100.0, 100.0);
+          layer.offset = FlutterPointMake(20.0, 20.0);
+
+          ASSERT_EQ(*layers[1], layer);
+        }
+
+        // Layer 2
+        {
+          FlutterBackingStore backing_store = *layers[2]->backing_store;
+          backing_store.type = kFlutterBackingStoreTypeOpenGL;
+          backing_store.did_update = true;
+          backing_store.open_gl.type = kFlutterOpenGLTargetTypeTexture;
+
+          FlutterLayer layer = {};
+          layer.struct_size = sizeof(layer);
+          layer.type = kFlutterLayerContentTypeBackingStore;
+          layer.backing_store = &backing_store;
+          layer.size = FlutterSizeMake(100.0, 100.0);
+          layer.offset = FlutterPointMake(20.0, 20.0);
```

Comment:
I'm probably confused but shouldn't this be 0,0?

The test code is: 
```dart
SceneBuilder builder = SceneBuilder();
    builder.pushOffset(0.0, 0.0);

     // 10
    builder.addPicture(Offset(10.0, 10.0), CreateColoredBox(red, size)); // red - flutter

     // 20
    builder.pushOffset(20.0, 20.0);
      builder.addPlatformView(1, width: size.width, height:size.height); // green - platform
    builder.pop();

     // 30
    builder.addPicture(Offset(30.0, 30.0), CreateColoredBox(blue, size)); // blue - flutter

```

the [20,20] offset is popped just before adding the layer with index 2.

Reply:
You're right. But, isn't the size wrong too? Shouldn't it be the frame size?

Reply:
Fixed. I also updated the render size.